### DISCUSSION TITLE
How to preview SAML Assertion attributes of particular users or users belonging to groups or roles?

### DISCUSSION THREAD

**Discussion by jgomo3** [keycloak/keycloak#35208] [Q&A]
Is there a tool where I could see what would be the `saml:AttributeStatement` that would be used in the SAML assertions for a particular user, for a role or a group?

Currently, I have to setup a test authentication flow where I capture the transferred XML and inspect it.  I think that is too much for simply inspecting that section.  There must be some kind of tool to preview it, or to preview the full SAML.

**Comment by embesozzi** [comment]
I find that the easiest way to troubleshoot SAML is by using a Chrome extension, such as SAML-tracer, to view SAML requests and responses

**Reply by jgomo3** [reply]
Yes, I'm beginning to accept that that is the only way :-)

I'm even having a hard time building a workflow with Postman, and wondering: why? Will that be so much better than the current workflow I have with my own app + those browser extensions/plugins that I already have and use in the first place?

**Comment by embesozzi** [comment]
For me, I ended up playing with the SAML extension and a custom app with SAML only for testing and demo purposes.
